raiden/raiden_service.py

```diff
@@ -873,29 +873,30 @@ def _initialize_payment_statuses(self, chain_state: ChainState):
         """
 
         with self.payment_identifier_lock:
-            for task in chain_state.payment_mapping.secrethashes_to_task.values():
-                if not isinstance(task, InitiatorTask):
-                    continue
-
-                # Every transfer in the transfers_list must have the same target
-                # and payment_identifier, so using the first transfer is
-                # sufficient.
-                initiator = next(iter(task.manager_state.initiator_transfers.values()))
-                transfer = initiator.transfer
-                transfer_description = initiator.transfer_description
-                target = transfer.target
-                identifier = transfer.payment_identifier
-                balance_proof = transfer.balance_proof
-                payment_hash_invoice = transfer.payment_hash_invoice
-                self.targets_to_identifiers_to_statuses[target][identifier] = PaymentStatus(
-                    payment_identifier=identifier,
-                    payment_hash_invoice=payment_hash_invoice,
-                    amount=transfer_description.amount,
-                    token_network_identifier=TokenNetworkID(
-                        balance_proof.token_network_identifier
-                    ),
-                    payment_done=AsyncResult(),
-                )
+            for node_address in chain_state.payment_mapping.keys():
```

Comment:
again you do not need `keys()` here, and you can get away with directly accessing the dict values instead